internal/metamorph/processor_helpers.go

```diff
@@ -9,14 +9,19 @@ func (p *Processor) GetProcessorMapSize() int {
 	return p.responseProcessor.getMapLen()
 }
 
-func updateStatusMap(statusUpdatesMap map[chainhash.Hash]store.UpdateStatus, statusUpdate store.UpdateStatus) {
+func (p *Processor) updateStatusMap(statusUpdatesMap map[chainhash.Hash]store.UpdateStatus, statusUpdate store.UpdateStatus) {
 	foundStatusUpdate, found := statusUpdatesMap[statusUpdate.Hash]
 
 	if !found || shouldUpdateStatus(statusUpdate, foundStatusUpdate) {
 		if len(statusUpdate.CompetingTxs) > 0 {
 			statusUpdate.CompetingTxs = mergeUnique(statusUpdate.CompetingTxs, foundStatusUpdate.CompetingTxs)
 		}
 
+		statusUpdate.StatusHistory = append(foundStatusUpdate.StatusHistory, store.StatusWithTimestamp{
+			Status:    foundStatusUpdate.Status,
+			Timestamp: p.now(),
+		})
+
```

Comment:
You can have a nil pointer exception here, because the if statement has also `!found` check. So you need to check if found and only then append